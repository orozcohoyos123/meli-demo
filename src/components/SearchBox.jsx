import React, { useState } from "react";
import PropTypes from "prop-types";
import { useHistory } from 'react-router-dom';
import SearchLogo from "../assets/ic_Search.png";
import './styles/SearchBox.scss';


const SearchBox = () => {
  const history = useHistory();
  const [query, setQuery] = useState("");
  
  const handleSubmit = (e) => {
    e.preventDefault();
        
    if (!query.trim()) return;
    
    history.push({
      pathname: '/items',
      search: `?search=${query.trim()}`
    })
  };

  return (
    <div className="searchBox col-9 col-md-10">
      <form onSubmit={handleSubmit}>
        <input 
          className="searchBox__input"
          name="query"
          type="text" 
          placeholder="Nunca dejes de buscar" 
          onChange={e => setQuery(e.target.value)}/>
        <button type="submit" className="searchBox__button">
          <img src={SearchLogo} alt="Search" />
        </button>
      </form>
    </div>
  );
};

SearchBox.propTypes = {
  onSubmit: PropTypes.func
};

export default SearchBox;
