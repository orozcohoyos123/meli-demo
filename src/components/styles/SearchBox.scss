@import '../../styles/CommonColors.scss';

.searchBox{
    padding: 0;
    position: relative;
}

.searchBox__input {
    border: 1px solid $light-gray;
    border-radius: 4px;
    height: 38px;
    position: relative;
    width: 100%;
    outline: none;
    padding: 0px 58px 0px 10px;
}

.searchBox__button {
    background-color: $light-gray;
    border: none;
    border-radius: 0 4px 4px 0;
    height: 36px;
    margin-top: 1px;
    outline: none;
    position: absolute;
    right: 1px;
    width: 40px;
}

