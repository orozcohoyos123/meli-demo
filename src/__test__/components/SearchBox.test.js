import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SearchBox from '../../components/SearchBox';

describe('<SearchBox />', () => {
    test('Debe renderizar el componente SearchBox.', () => {
        const searchBox = mount(<SearchBox />);
        const input = searchBox.find('.searchBox__input');
        const button = searchBox.find('.searchBox__button');

        expect(searchBox.length).toEqual(1);
        expect(input.length).toEqual(1);
        expect(button.length).toEqual(1);
    }) 

    test('No debe realizar el evento submit si no llena el input.', () => {
        const handleSubmit = jest.fn();
        const wrapper = mount(<SearchBox onSubmit={handleSubmit}/>);
        const form = wrapper.find('form');

        form.simulate('submit')
        expect(handleSubmit).toHaveBeenCalledTimes(0); 
    })   

    test('Debe realizar el evento submit si llena el input.', () => {
        const handleSubmit = jest.fn();
        const wrapper = mount(
            <BrowserRouter>
                <SearchBox onSubmit={handleSubmit}/>
            </BrowserRouter>
        );
        
        const form = wrapper.find('form');

        wrapper.find('input').simulate('change', { target: { value: 'Ps4' } });
        form.simulate('submit')
        expect(handleSubmit).toHaveBeenCalled(); 
    }) 
});

describe('SearchBox sanpshot', () => { 
    test('Comprobar UI componente SearchBox', () => {
        const searchBox = create(<SearchBox />);
        expect(searchBox.toJSON()).toMatchSnapshot();
    })
});
