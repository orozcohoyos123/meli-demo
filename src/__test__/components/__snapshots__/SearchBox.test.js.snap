// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox sanpshot Comprobar UI componente SearchBox 1`] = `
<div
  className="searchBox col-9 col-md-10"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="searchBox__input"
      name="query"
      onChange={[Function]}
      placeholder="Nunca dejes de buscar"
      type="text"
    />
    <button
      className="searchBox__button"
      type="submit"
    >
      <img
        alt="Search"
        src="ic_Search.png"
      />
    </button>
  </form>
</div>
`;
